### PR TITLE
[Enhanced Http] Support for http proxying scenarios

### PR DESCRIPTION
**Pull request by satvu**
### Issue describing the changes in this PR

Add initial components for the enhanced HTTP model (read more [here](https://github.com/Azure/azure-functions-dotnet-worker/issues/1387)). These components are hidden behind a capability and this code path should only light up when a worker implementing the HTTP model provides the necessary info.

These are minimal changes needed to make it easier for partner teams and stakeholders to use and provide feedback as we continue to develop this feature.

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Comment by satvu** [comment]
Working on tests but the rest is ready for review.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 1]
```diff
@@ -420,6 +426,14 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
                 _isWorkerApplicationInsightsLoggingEnabled = true;
             }
 
+            // If http proxying is enabled, we need to get the proxying endpoint of this worker
+            var httpUri = _workerCapabilities.GetCapabilityState(RpcWorkerConstants.HttpUri);
+            if (!string.IsNullOrEmpty(httpUri))
+            {
+                _httpProxyEndpoint = new Uri(httpUri);
+                _isHttpProxyingWorker = true;
```
Comment (jviau) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: Can this be a property based on `_httpProxyEndpoint`?

``` CSharp
private bool IsHttpProxyingWorker => _httpProxyEndpoint is not null;
```

File: src/WebJobs.Script.Grpc/Server/IHttpProxyService.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Yarp.ReverseProxy.Forwarder;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc
+{
+    public interface IHttpProxyService
+    {
+        public ValueTask<ForwarderError> Forward(ScriptInvocationContext context, Uri httpUri);
```
Comment (jviau) [inline src/WebJobs.Script.Grpc/Server/IHttpProxyService.cs]: nit: 
```suggestion
        public ValueTask<ForwarderError> ForwardAsync(ScriptInvocationContext context, Uri httpUri);
```

Also, should this take in a `CancellationToken`? Or is one available on the context object?

File: src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Yarp.ReverseProxy.Forwarder;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc
+{
+    internal class DefaultHttpProxyService : IHttpProxyService, IDisposable
+    {
+        private SocketsHttpHandler _handler;
```
Comment (jviau) [inline src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs]: nit: please name all of these fields `readonly` (unless they need re-assignment).

File: src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        private IHttpForwarder _httpForwarder;
+        private HttpMessageInvoker _messageInvoker;
+        private ForwarderRequestConfig _forwarderRequestConfig;
+
+        public DefaultHttpProxyService(IHttpForwarder httpForwarder)
+        {
+            _httpForwarder = httpForwarder;
```
Comment (jviau) [inline src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs]: nit: check for null. I think being .net6 you have these two ways to do it:

``` csharp
ArgumentNullException.ThrowIfNull(httpForwarder);
_httpForwarder = httpForwarder
```

or

``` csharp
_httpForwader = httpForwarder ?? throw new ArgumentNullException(nameof(httpForwarder);
```

File: src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+        public void Dispose()
+        {
+            _handler?.Dispose();
+            _messageInvoker?.Dispose();
+        }
+
+        public ValueTask<ForwarderError> Forward(ScriptInvocationContext context, Uri httpUri)
+        {
+            if (context is null)
```
Comment (jviau) [inline src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs]: I think we are .net6 in this project right? If so, you can shorten this to:

``` csharp
ArgumentNullException.ThrowIfNull(context);
```

File: src/WebJobs.Script/ScriptConstants.cs [hunk 6]
```diff
@@ -225,5 +225,9 @@ public static class ScriptConstants
         public static readonly string LiveLogsSessionAIKey = "#AzFuncLiveLogsSessionId";
 
         public static readonly string FunctionsHostingConfigSectionName = "FunctionsHostingConfig";
+
+        // HTTP Proxying constants
+        public static readonly string IsHttpProxyingEnabled = "IsHttpProxying";
+        public static readonly string HttpProxyCorrelationHeader = "x-invocation-id";
```
Comment (jviau) [inline src/WebJobs.Script/ScriptConstants.cs]: Going by other extension headers, I think we want `x-ms-` as the prefix.
```suggestion
        public static readonly string HttpProxyCorrelationHeader = "x-ms-invocation-id";
```

File: test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json [hunk 7]
```diff
@@ -2190,11 +2190,11 @@
           "System.Runtime": "4.3.1"
         }
       },
-      "System.IO.Hashing/6.0.0": {
```
Comment (satvu) [inline test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json:2193]: https://www.fuget.org/packages/System.IO.Hashing/7.0.0/lib/netstandard2.0/diff/6.0.0/

Reviewed the above w/ @fabiocav we believe this is a safe change.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 8]
```diff
@@ -420,6 +426,14 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
                 _isWorkerApplicationInsightsLoggingEnabled = true;
             }
 
+            // If http proxying is enabled, we need to get the proxying endpoint of this worker
+            var httpUri = _workerCapabilities.GetCapabilityState(RpcWorkerConstants.HttpUri);
+            if (!string.IsNullOrEmpty(httpUri))
+            {
+                _httpProxyEndpoint = new Uri(httpUri);
```
Comment (surgupta-msft) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: We should check if `httpUri` has the correct format and also handle cases where `new Uri(httpUri)` can throw exceptions.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs (hunk 1 above)
Comment (surgupta-msft) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: Do we need these 2 variables at global level and in this function? Can I not directly check for capability and assign _httpProxyEndpoint in `SendInvocationRequest()`?

Nit - If variable `_isHttpProxyingWorker` is needed - not sure if "proxying" is appropriate to use. In case we want to change it, some alternatives are - `_WorkerSupportsHttpProxy` or `_IsWorkerHttpProxyEnabled` or something similar.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 9]
```diff
@@ -420,6 +426,14 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
                 _isWorkerApplicationInsightsLoggingEnabled = true;
             }
 
+            // If http proxying is enabled, we need to get the proxying endpoint of this worker
+            var httpUri = _workerCapabilities.GetCapabilityState(RpcWorkerConstants.HttpUri);
```
Comment (surgupta-msft) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:430]: nit - Are we checking somewhere else if HttpProxying is enabled and then fetching `RpcWorkerConstants.HttpUri`? The comment above is bit confusing for me. If `HttpUri` is acting as a capability, then may be we can rename it to `SupportsHttpUri` or `HandlesHttpUri` or `SupportsHttpProxing` or something similar.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 10]
```diff
@@ -738,6 +752,13 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
                 {
                     context.CancellationToken.Register(() => SendInvocationCancel(invocationRequest.InvocationId));
                 }
+
+                if (_isHttpProxyingWorker && context.FunctionMetadata.IsHttpTriggerFunction())
+                {
+                    var aspNetTask = _httpProxyService.Forward(context, _httpProxyEndpoint);
+
+                    context.Properties.Add("HttpProxyingTask", aspNetTask);
```
Comment (surgupta-msft) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: Defining constant for string literal `HttpProxyingTask`

File: src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (context.Inputs is null)
```
Comment (surgupta-msft) [inline src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs]: Should we also add Empty check here?

File: src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+            {
+                throw new InvalidOperationException($"Cannot proxy the HttpTrigger function {context.FunctionMetadata.Name} without an input of type {nameof(HttpRequest)}.");
+            }
+
+            HttpContext httpContext = httpRequest.HttpContext;
+
+            httpContext.Items.Add(ScriptConstants.IsHttpProxyingEnabled, bool.TrueString);
+
+            // add invocation id as correlation id
+            httpRequest.Headers.TryAdd(ScriptConstants.HttpProxyCorrelationHeader, context.ExecutionContext.InvocationId.ToString());
```
Comment (surgupta-msft) [inline src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs:59]: We will need null check if `ExecutionContext` or `InvocationId` can be null here. Ignore if it's not necessary or being already checked somewhere else in the flow.

File: src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+
+            var aspNetTask = _httpForwarder.SendAsync(httpContext, httpUri.ToString(), _messageInvoker, _forwarderRequestConfig);
```
Comment (surgupta-msft) [inline src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs:61]: This line can throw exception if arguments (especially httpUri) are not in the expected format. We should also add tests for such unhappy scenarios.

File: src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs [hunk 14]
```diff
@@ -52,6 +53,15 @@ public async Task Invoke(HttpContext context)
 
                 int nestedProxiesCount = GetNestedProxiesCount(context, functionExecution);
                 IActionResult result = await GetResultAsync(context, functionExecution);
+
+                if (context.Items.TryGetValue(ScriptConstants.IsHttpProxyingEnabled, out var value))
+                {
+                    if (value.ToString() == bool.TrueString)
```
Comment (surgupta-msft) [inline src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs]: can `value` be null here? Thinking of a case where it is not added in ScriptConstants but being checked here.

File: src/WebJobs.Script/Extensions/FunctionMetadataExtensions.cs [hunk 15]
```diff
@@ -24,15 +26,20 @@ public static bool IsHttpInAndOutFunction(this FunctionMetadata metadata)
...
             {
                 return true;
             }
 
             return false;
         }
 
+        public static bool IsHttpTriggerFunction(this FunctionMetadata metadata)
+        {
+            return metadata.InputBindings.Any(b => string.Equals(HttpTriggerKey, b.Type, StringComparison.OrdinalIgnoreCase));
```
Comment (surgupta-msft) [inline src/WebJobs.Script/Extensions/FunctionMetadataExtensions.cs:40]: null check for metadata and InputBindings.

File: src/WebJobs.Script/ScriptConstants.cs [hunk 16]
```diff
@@ -225,5 +225,9 @@ public static class ScriptConstants
         public static readonly string LiveLogsSessionAIKey = "#AzFuncLiveLogsSessionId";
 
         public static readonly string FunctionsHostingConfigSectionName = "FunctionsHostingConfig";
+
+        // HTTP Proxying constants
+        public static readonly string IsHttpProxyingEnabled = "IsHttpProxying";
```
Comment (surgupta-msft) [inline src/WebJobs.Script/ScriptConstants.cs]: nit - `HttpProxyEnabled` or `SupportsHttpProxy` since it's a string

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 17]
```diff
@@ -81,6 +82,9 @@ internal class GrpcWorkerChannel : IRpcWorkerChannel, IDisposable
         private bool _isSharedMemoryDataTransferEnabled;
         private bool? _cancelCapabilityEnabled;
         private bool _isWorkerApplicationInsightsLoggingEnabled;
+        private IHttpProxyService _httpProxyService;
+        private Uri _httpProxyEndpoint;
+        private bool _isHttpProxyingWorker = false;
```
Comment (soninaren) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: This would be cached when the worker is running under placeholder mode. Likely may not matter for dotnet worker. However, if there is customer config that controls the behavior then we may run into caching issues.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs (hunk 1 above)
Comment (satvu) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: Regarding names it was discussed in the PR into the feature branch (which is now this PR into dev) and we will change them once we decide on final names. Mention of "proxy" is probably only useful internally so this is up for debate still.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs (hunk 9 above)
Comment (satvu) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:430]: We are calling it this because the working sends a Uri (string form) over instead of `bool.TrueString`

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 18]
```diff
@@ -420,6 +426,21 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
                 _isWorkerApplicationInsightsLoggingEnabled = true;
             }
 
+            // If http proxying is enabled, we need to get the proxying endpoint of this worker
+            var httpUri = _workerCapabilities.GetCapabilityState(RpcWorkerConstants.HttpUri);
+            if (!string.IsNullOrEmpty(httpUri) && FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagEnableHttpProxying))
```
Comment (soninaren) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:431]: `FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagEnableHttpProxying)` will always be false for worker started in placeholder mode

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 19]
```diff
@@ -738,6 +759,13 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
                 {
                     context.CancellationToken.Register(() => SendInvocationCancel(invocationRequest.InvocationId));
                 }
+
+                if (_isHttpProxyingWorker && context.FunctionMetadata.IsHttpTriggerFunction())
```
Comment (soninaren) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: `IsHttpTriggerFunction` explicitly looks at httpTrigger functions only. This does not include any webhooks from extensions such as durable or trigger that use http such as eventGrid. I am assuming this is by design.

File: src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs (hunk 13 above)
Comment (satvu) [reply src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs:61]: Since `httpUri` is passed in as a paremter (of type Uri), the result of `toString()` should be valid.

File: src/WebJobs.Script.Grpc/GrpcServiceCollectionsExtensions.cs [hunk 20]
```diff
@@ -16,6 +16,9 @@ public static IServiceCollection AddScriptGrpc(this IServiceCollection services)
 
             services.AddSingleton<IRpcServer, AspNetCoreGrpcServer>();
 
+            services.AddHttpForwarder();
```
Comment (satvu) [inline src/WebJobs.Script.Grpc/GrpcServiceCollectionsExtensions.cs:19]: Looking for a clean way to avoid adding this without a feature flag.

File: src/WebJobs.Script.Grpc/GrpcServiceCollectionsExtensions.cs [hunk 21]
```diff
@@ -16,6 +16,9 @@ public static IServiceCollection AddScriptGrpc(this IServiceCollection services)
...
+            services.AddSingleton<IHttpProxyService, DefaultHttpProxyService>();
```
Comment (satvu) [inline src/WebJobs.Script.Grpc/GrpcServiceCollectionsExtensions.cs:20]: Unfortunately, we can't get this behind the feature flag (can't access `FeatureFlags` yet at this point, host will crash) so these services will be added but not used.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 22]
```diff
@@ -902,6 +930,17 @@ internal async Task InvokeResponse(InvocationResponse invokeResponse)
             {
                 if (invokeResponse.Result.IsInvocationSuccess(context.ResultSource, capabilityEnabled))
```
Comment (soninaren) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:930]: What happens if `IsInvocationSuccess` is false but the proxy request made it through. Also what happens when `IsInvocationSuccess` is false and proxy request has failed.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 23]
```diff
@@ -738,6 +759,13 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
                 {
                     context.CancellationToken.Register(() => SendInvocationCancel(invocationRequest.InvocationId));
                 }
+
+                if (_isHttpProxyingWorker && context.FunctionMetadata.IsHttpTriggerFunction())
+                {
+                    var aspNetTask = _httpProxyService.Forward(context, _httpProxyEndpoint);
```
Comment (soninaren) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: Are there any down sides of awaiting the task here? From what i am understanding the proxy request is already made here and awaiting the task simply provides with the forward errors if any. we can store the proxy result in the context and log it along with invocation success of failure.

Comment (satvu) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: Original design awaited the task here - I believe it changed because this task often completes after function invocation due to logic on the worker side (if there are no errors, this awaits until the http response is written back) and we decided to just send it off and unify if the function invocation succeeds.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs (hunk 22 above)
Comment (satvu) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:930]: The design for these cases hasn't been discussed but for now if `IsInvocationSuccess` fails at all we will ignore any proxying results and log an error.

File: test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json (hunk 7 above)
Comment (jviau) [reply test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json:2193]: Curious why this file is included at all in our code base? Being a build/restore artifact, shouldn't it be excluded?

File: src/WebJobs.Script.Grpc/Server/IHttpProxyService.cs (hunk 2 above)
Comment (satvu) [reply src/WebJobs.Script.Grpc/Server/IHttpProxyService.cs]: @jviau Yes `CancellationToken` implementation is something on the to-do list across the entire flow between host and worker.

**Comment by satvu** [comment]
Closing and postponing to next deadline instead.

File: src/WebJobs.Script.Grpc/Server/IHttpProxyService.cs [hunk 24]
```diff
@@ -0,0 +1,15 @@
...
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Yarp.ReverseProxy.Forwarder;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc
+{
+    public interface IHttpProxyService
+    {
+        public ValueTask<ForwarderError> ForwardAsync(ScriptInvocationContext context, Uri httpUri);
```
Comment (jviau) [inline src/WebJobs.Script.Grpc/Server/IHttpProxyService.cs]: small nit: don't need to include `public` on interfaces, they are public by default.
```suggestion
        ValueTask<ForwarderError> ForwardAsync(ScriptInvocationContext context, Uri httpUri);
```

**Review by soninaren — APPROVED** [review]

**Comment by satvu** [comment]
Noting again for the record this is for _private preview_ and reopening again because we want to reemphasize the main concerns:

- are we concerned about reliability?
- are we concerned about our ability to iterate

If both are no, then this work should be fine for now. We want to put this in the hands of some PMs and ASP.NET partners for further validation.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 25]
```diff
@@ -420,6 +426,20 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
                 _isWorkerApplicationInsightsLoggingEnabled = true;
             }
 
+            // If http proxying is enabled, we need to get the proxying endpoint of this worker
+            var httpUri = _workerCapabilities.GetCapabilityState(RpcWorkerConstants.HttpUri);
```
Comment (jviau) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:430]: Should the feature flag check be moved around _all_ of the HTTP proxy logic?

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 26]
```diff
@@ -902,6 +929,17 @@ internal async Task InvokeResponse(InvocationResponse invokeResponse)
             {
                 if (invokeResponse.Result.IsInvocationSuccess(context.ResultSource, capabilityEnabled))
                 {
+                    if (context.Properties.TryGetValue(ScriptConstants.HttpProxyTask, out ValueTask<ForwarderError> httpProxyTask))
```
Comment (jviau) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: Storing value task like this is _slightly_ uncomfortable to me. ValueTask is meant to be a short-lifetime task that **does not support** awaiting multiple times. Depending on the `ValueTask` itself, it might not even be safe to set aside and unwrap later (not entirely sure about this, but looking at `IValueTaskSource` I could see that being potentially true). Bit of a read here: https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/

You _might_ have to call `ValueTask.AsTask()` and store that in properties, just to be safe. Although if we can review if the two stage invoke and later await is necessary in this path would be best (but we can also review that later).

**Review by jviau — APPROVED** [review]

**Comment by jviau** [comment]
So long as it is clear this is private preview and we are okay with making breaking changes later I am fine with this.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs (hunk 25 above)
Comment (satvu) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:430]: The other logic is accessed through the _IsHttpProxyingWorker_ var which can only be set to true here when the feature flag is checked - I can do that as well but I was concerned about excessive checking since it would occur in every invocation + invocation response.

Comment (satvu) [reply src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs:430]: Discussed w/ Naren we think it's cheap enough and the extra protection is good

File: test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json (hunk 7 above)
Comment (satvu) [reply test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json:2193]: Skipped this comment earlier but I am only editing it as it's needed to pass tests - it's a file in the test repo to make sure all dependency changes are caught and approval is asked from specific code owners (the test error message says so, you can see in earlier pipeline failures).

**Review by fabiocav — APPROVED** [review]
